docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,747 @@
+Motivation
+----------
+
+Currently, Open Enclave provides several APIs that developers can use for
+attestation. The two key functions are `oe_get_report`, which produces an opaque
+blob that is signed by the enclave, and `oe_verify_report`, which can be used to
+verify the generated report. The original purpose of those two APIs were to
+provide a simple, cross-platform way to produce and verify attestation data.
+
+However, some developers need more flexibility for their attestation
+requirements. For example, one might want to extend Open Enclave's
+current attestation structures with extra information, such as geolocation
+or a timestamp. Another user might want their enclaves to generate attestation
+data that is in a compatible format with their existing authentication
+infrastructure, such as a JSON Web Token or an X.509 certificate. There are also
+users who want to specify their endorsements (information from a second source
+used for verification), instead of using the set of endorsements provided by Open
+Enclave.
+
+Overall, there has been interest in enhancing Open Enclave's APIs to support
+custom attestation formats to enable these scenarios.
+
+Terminology
+-----------
+
+This document uses the following terminology defined below. Note that
+these definitions are consistent with the terms defined in the
+[Remote Attestation Procedures (RATS)](https://datatracker.ietf.org/wg/rats/about/)
+working group.
+
+- Claims
+  - Claims are statements about a particular subject. They consist of
+    name-value pairs containing the claim name, which is a string, and
+    claim value, which is arbitrary data. Example of claims could be
+    [name="version", value=1] or [name="enclave_id", value=1111].
+- Evidence
+  - This is the data about the enclave that is produced and signed by it.
+    The SGX report would be an example of evidence.
+- Endorsements
+  - This is additional data that used in the evidence verification process,
+    but is not produced by the enclave. An example of an endorsement would be
+    the quoting enclave's identity used in SGX remote attestation, because it
+    is retrieved from Intel's servers, rather than the enclave.
+- Verifier
+  - The verifier is responsible for taking in the evidence and endorsements
+    and deciding if the enclave is trustworthy.
+- Relying party
+  - The relying party is the entity interested in communicating with an
+    enclave. The enclave must attest to the relying party before the
+    relying party can trust it. The relying party can also play the role
+    of the verifier, but it does not necessarily have to.
+
+Specification
+-------------
+
+To support custom attestation formats, this document proposes adding a plugin
+model for attestation. The Open Enclave SDK will define a set of common APIs
+that each plugin must implement. Each plugin will define an UUID to distinguish
+it from other plugins.
+
+Futhermore, there will be additional attestation "plugin aware" APIs that are
```

Comment:
Maybe mention the functions that we agreed with, namely oe_get_evidence() and oe_verify_evidence()

Reply:
Agree.
Let's add links to the "Remote Attestation Endorsements" design documentation for those two functions, which would make the whole picture more clear.

Reply:
Mentioned function and added a link to Sergio's doc.

Reply:
@jazzybluesea The endorsement is still named `RemoteAttestationCollaterals.md` instead of `RemoteAttestationEndorsements.md`.